docs/dev/tombstone.md

```diff
@@ -0,0 +1,572 @@
+To determine whether an expired cell is really live or not, one has to look at the `expiry` and compare it with the current time.
+If `expiry <= now()`, the cell is expired and it will be treated as a dead cell by compaction.
+
+Reads also compact the data, so if we read this partition, the cell will not show up in the results:
+```CQL
+SELECT * FROM ks.tbl WHERE pk = 'expired cell';
+
+ pk | ck1 | ck2 | v1 | v2
+----+-----+-----+----+----
+
+(0 rows)
+```
+
+After flushing and compacting the table, the expired cell is now converted to a cell tombstone:
+```CQL
+SELECT * FROM MUTATION_FRAGMENTS(ks.tbl) WHERE pk = 'expired cell';
+
+ pk           | mutation_source                                                                                                  | partition_region | ck1 | ck2 | position_weight | metadata                                                                                                                                                          | mutation_fragment_kind | value
+--------------+------------------------------------------------------------------------------------------------------------------+------------------+-----+-----+-----------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------+-------------
+ expired cell | sstable:/var/lib/scylla/data/ks/tbl-b0f65c600ad611f0811d8550c417dfcd/me-3gox_0jgn_1gq5c20k31rxg89le0-big-Data.db |                0 |     |     |                 |                                                                                                                                                  {"tombstone":{}} |        partition start |        null
+ expired cell | sstable:/var/lib/scylla/data/ks/tbl-b0f65c600ad611f0811d8550c417dfcd/me-3gox_0jgn_1gq5c20k31rxg89le0-big-Data.db |                2 |   0 |   0 |               0 | {"marker":{"timestamp":1743058565262883},"columns":{"v1":{"is_live":false,"type":"regular","timestamp":1743058565262883,"deletion_time":"2025-03-27 06:56:05z"}}} |         clustering row | {"v1":null}
+ expired cell | sstable:/var/lib/scylla/data/ks/tbl-b0f65c600ad611f0811d8550c417dfcd/me-3gox_0jgn_1gq5c20k31rxg89le0-big-Data.db |                3 |     |     |                 |                                                                                                                                                              null |          partition end |        null
+
+(3 rows)
+```
+
+## Row Marker
+
+The row marker is an object that is part of a the row.
+It stores no data, but it has a timestamp, it can have a TTL and it also interacts with tombstones, just like data.
+The row marker has a special interaction with the [shadowable tombstone](./tombstone.md#shadowable-tombstone).
+This special interaction will be examined in detail in the next chapter.
+
```

Comment:
There is also interaction of the row marker with SELECT (the primary interaction).

SELECT of a row with a live row marker and no live cells will display the row (with all regular columns NULL).
SELECT of a row with no live row marker and no live cells will not display the row.

Reply:
Oh you do say it. But I think it needs to be said here.

Reply:
Done in v2.